README.md

```diff
@@ -0,0 +1,743 @@
+
+Why would you need such extremely fine-grained events? The applications are diverse and surprisingly common:
+
+- Automatically attaching *and* detaching main page elements in response to route changes. One possible way to approach this use case is illustrated in the example code below.
+- Responding to the different stages of a user's sign-in/sign-out process.
+- Tracking the various editing stages of a form element (pristine, focused, editing, tainted, valid, invalid, etcetera).
+- Tracking whether a server is reachable; you typically want to respond very differently to a server going *offline* than to a server returning *online*.
+- Your imagination is the limit. 😊
+
+> In case this reminds you of [finite-state machines][fsm]: there is, indeed, a strong overlap in functionality, hence the name "state model". This module was inspired by (and written by the same author as) [backbone-machina][bb-mach].
+
+``` javascript
+import { Model, Router, history } from 'backbone';
+import { getStateMixin } from '@uu-cdh/backbone-util';
+import { homeView, profileView, settingsView } from './your/own/code';
+
+// Obtain the mixin.
+var stateMixin = getStateMixin();
+// Make a class using the mixin.
+var State = Model.extend(stateMixin);
+// Get an instance, initially empty. We will use this below.
+var state = new State;
+
+// Our router. We don't need methods, just names.
+var router = new Router({
+    '': 'home',
+    'home': 'home',
+    'profile': 'profile',
+    'settings': 'settings'
+});
+
+// Now, we use events to declaratively list which main view
+// *should* be on the page, depending on the route.
+router.on({
+    'route:home': () => state.set('mainView', homeView),
+    'route:profile': () => state.set('mainView', profileView),
+    'route:settings': () => state.set('mainView', settingsView),
+});
+history.on('notfound', () => state.unset('mainView'));
+
+// Finally, we use the fine-grained state events to attach the
+// new main view (if there is one) and detach the previous main
+// view (if there is one). Note how we don't need any if-else
+// decision trees and the events ensure consistency. Also, these
+// two event bindings will be sufficient even if we have 100
+// different routes and main views!
+state.on({
+    'exit:mainView': (s, view) => view.$el.detach(),
+    'enter:mainView': (s, view) => view.$el.appendTo('main'),
+});
+```
+
+[fsm]: https://en.wikipedia.org/wiki/Finite-state_machine
+[bb-mach]: https://www.npmjs.com/package/backbone-machina
+
+#### Function `getStateMixin`
+
+**Default export** of `@uu-cdh/backbone-util/src/state-model.js`, **reexported by name** from the package index.
+
+**Parameter:** `options`, an optional control value to customize the mixin. If not omitted, it can be a boolean or a (plain) object with any properties that you want to add to the mixin.
```

Comment:
Good point, I'll bring that explanation forward.

Passing `true` and an object is equivalent to only passing an object. One might theoretically pass `false` and an object to prevent the default `preinitizalize` from being included, while adding one's own `initialize` or `constructor` that includes a call to `this.bindStateEvents`. The added value of that is questionable, because you can save yourself the effort of writing that line by simply using the default `preinitizalize`.